File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        opns = 0

        i = 0
        while opns <= len(nums) and i < len(nums):
            opns += 1
            if nums[i] == 0:
                nums.pop(i)
                nums.append(0)
            else:
                i += 1

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int i = 0;

        for(int j = 1; j < nums.size(); j++){
            if(nums[i] == 0 && nums[j] != 0){
                //swap
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            }
            else if(nums[i] == nums[j] and nums[j] == 0){
                i -= 1;
            }
            i++;
        }

    }
};